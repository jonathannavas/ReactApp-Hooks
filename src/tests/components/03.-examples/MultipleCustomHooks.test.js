import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03.-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');


describe('Pruebas en <MultipleCustomHooks />',()=>{

    beforeEach( () => {
        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {}
        })
    })

    test('Debe de mostrarse correctamente', () => {
        
        useFetch.mockReturnValue({
            data:null,
            loading:false,
            error: null
        })
        const wrapper = shallow( <MultipleCustomHooks /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar la informacion', () => {
       
        useFetch.mockReturnValue({
            data:[{
                author: 'Jonathan',
                quote: 'Hola mundo'
            }],
            loading:false,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHooks /> );

        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect( wrapper.find('.mb-3').text().trim() ).toBe( 'Hola mundo' );
        expect( wrapper.find('footer').text().trim() ).toBe( 'Jonathan' );


        
    });
    



});