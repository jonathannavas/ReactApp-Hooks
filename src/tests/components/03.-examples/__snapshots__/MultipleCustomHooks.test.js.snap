// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Breaking Quotes
  </h1>
  <hr />
  <blockquote
    className="blockquote text-end"
  >
    <p
      className="mb-3"
    >
       
       
    </p>
    <footer
      className="blockquote-footer"
    >
       
    </footer>
  </blockquote>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente
  </button>
</div>
`;
